roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -110,5 +105,6 @@
         cifmw_libvirt_manager_mac_map: >-
           {{
             cifmw_libvirt_manager_mac_map | default({}) |
-            combine(_dataset, recursive=true)
+            combine(_dataset, list_merge='append', recursive=true)
           }}
+        cacheable: true
```

Comment:
I don't think we should cache it. We'd better check if the file exists, and load it (the `cifmw_libvirt_manager_mac_map` fact is set in some other place, iirc `deploy_layout.yml`)
Something like this should do:
```YAML
- name: load interfaces-info.yml if exists
 register: _interfaces_info
  when:
    - interfaces-info.yml is exists  # yes, "is exists" is correct
  ansible.builtin.slurp:
    path: interfaces-info.yml

- name: Initialize fact
  vars:
    _loaded: "{{ _interface_info.content | b64decode | default('{}') | from_yaml }}"
  ansible.builtin.set_fact:
    cifmw_libvirt_manager_mac_map: "{{ _loaded }}"
```

That would allow to extend the list in case we're adding a compute node during a second run.

Reply:
Yes, we shouldn't be caching them. There is one peculiar issue that we are seeing at our end.